File: citadel-first/mobile/lib/models/face_verification.dart
class FaceVerificationResult {
  final int id;
  final int appUserId;
  final bool isMatch;
  final double confidence;
  final double? distance;
  final bool selfieFaceDetected;
  final bool docFaceDetected;
  final String message;

  const FaceVerificationResult({
    required this.id,
    required this.appUserId,
    required this.isMatch,
    required this.confidence,
    this.distance,
    required this.selfieFaceDetected,
    required this.docFaceDetected,
    required this.message,
  });

  factory FaceVerificationResult.fromJson(Map<String, dynamic> json) =>
      FaceVerificationResult(
        id: json['id'] as int,
        appUserId: json['app_user_id'] as int,
        isMatch: json['is_match'] as bool,
        confidence: (json['confidence'] as num).toDouble(),
        distance: (json['distance'] as num?)?.toDouble(),
        selfieFaceDetected: json['selfie_face_detected'] as bool,
        docFaceDetected: json['doc_face_detected'] as bool,
        message: json['message'] as String,
      );
}

class FaceDetectResult {
  final bool faceDetected;
  final int faceCount;
  final String message;

  const FaceDetectResult({
    required this.faceDetected,
    required this.faceCount,
    required this.message,
  });

  factory FaceDetectResult.fromJson(Map<String, dynamic> json) => FaceDetectResult(
        faceDetected: json['face_detected'] as bool,
        faceCount: json['face_count'] as int,
        message: json['message'] as String,
      );
}

File: citadel-first/backend/app/schemas/face_verification.py
from pydantic import BaseModel, Field


class SelfiePresignedUrlRequest(BaseModel):
    filename: str
    content_type: str = "image/jpeg"


class SelfiePresignedUrlResponse(BaseModel):
    upload_url: str
    key: str


class FaceVerifyRequest(BaseModel):
    selfie_image_key: str
    doc_image_key: str


class FaceVerifyResponse(BaseModel):
    id: int
    app_user_id: int
    is_match: bool
    confidence: float = Field(ge=0.0, le=1.0)
    distance: float | None = None
    selfie_face_detected: bool
    doc_face_detected: bool
    message: str

    model_config = {"from_attributes": True}


class FaceDetectRequest(BaseModel):
    selfie_image_key: str


class FaceDetectResponse(BaseModel):
    face_detected: bool
    face_count: int = 0
    message: str